### PR TITLE
[WIP] Make generated CMake project independent from the project generator

### LINKED ISSUE #88: Generate clean CMake project
Currently, the generate CMake project is incomplete.
`simulate.py` is located in the build directory, but it is necessary to build the project.

I worked around the issue by appending the following to `hello-world/CMakeLists.txt`.
```
set(SEL4_TUTORIALS_DIR ../projects/sel4-tutorials)
set(TUT_BOARD pc)
set(TUT_ARCH x86_64)
set(TUTORIAL_DIR hello-world)
```

### PR DESCRIPTION
**Pull request by locriacyber**
Hope to close #88.

- [ ] set SEL4_TUTORIALS_DIR in top level when project is generated
- [ ] auto get TUTORIAL_DIR
- [ ] remove refresh madness (good for developers of tutorials, but should be turned off otherwise)

### CODE REVIEW

File: settings.cmake [hunk 1]
```diff
@@ -20,21 +20,17 @@ set(POLLY_DIR ${project_dir}/tools/polly CACHE INTERNAL "")
...
 include(application_settings)
 
-# Deal with the top level target-triplet variables.
-if(NOT TUT_BOARD)
-    message(
-        FATAL_ERROR
-            "Please select a board to compile for. Choose either pc or zynq7000\n\t`-DTUT_BOARD=<PREFERENCE>`"
-    )
-endif()
+set(TUT_BOARD "pc" CACHE STRING "Target machine to build the project for")
```
Comment (axel-h) [inline settings.cmake:23]: I think it is still useful to tell the user about the default choice that are made and how these can be changed:
```
if(NOT TUT_BOARD)
    message("No board selected (-DTUT_BOARD=[pc|zynq7000]), defaulting to 'pc'")
    set(TUT_BOARD "pc" CACHE STRING "Target machine to build the project for")
endif()
```

Comment (locriacyber) [reply settings.cmake]: done

**Comment by locriacyber** [comment]
BTW, I got a ready-to-go git repo for sel4 tutorials with none of the `repo` weirdness. Maybe this should be the preferred way to get tutorials?
https://github.com/locriacyber/seL4-tutorials-template

**Comment by axel-h** [comment]
@lsf37: how are the rules about the DCO, is `74560659+locriacyber@users.noreply.github.com` acceptable?

**Comment by axel-h** [comment]
> BTW, I got a ready-to-go git repo for sel4 tutorials with none of the `repo` weirdness. Maybe this should be the preferred way to get tutorials? https://github.com/locriacyber/seL4-tutorials-template

Thanks. While I'm also more a fan of submodules, I fear the issue about repo vs. submodules (vs. yet another solution) will never be resolved. Both have their advantages and disadvantages and both work with the proper instructions. With 'repo' being a 3rd party tool, there might be some resistance to use it. Another way to just distribute the tutorials would be a flattened well tested release package, that you just unzip/untar and use. Drawback here is that this makes upstreaming changes a bit harder because the git context is lost.
In the end it take somebody to maintain things. For the repo XML files there are scripts in the github CI workflow. This could be duplicated for submodules, but it take somebody to drive this. I don't really see a problem having both in parallel eventually.

File: Findsel4-tutorials.cmake [hunk 2]
```diff
@@ -58,4 +58,4 @@ macro(sel4_tutorials_setup_capdl_tutorial_environment)
 endmacro()
 
 include(FindPackageHandleStandardArgs)
-FIND_PACKAGE_HANDLE_STANDARD_ARGS(camkes-tool DEFAULT_MSG SEL4_TUTORIALS_DIR)
```
[the remark this answers is not in the document]
Comment (axel-h) [reply Findsel4-tutorials.cmake:61]: no need to make another branch or PR, just split up the commit. The idea is having one commit for one thing.

**Comment by locriacyber** [comment]
@axel-h How is `CMakeLists.txt` for hello-world generated?
`./init --tut hello-world`

**Comment by lsf37** [comment]
> > BTW, I got a ready-to-go git repo for sel4 tutorials with none of the `repo` weirdness. Maybe this should be the preferred way to get tutorials? https://github.com/locriacyber/seL4-tutorials-template
> 
> Thanks. While I'm also more a fan of submodules, I fear the issue about repo vs. submodules (vs. yet another solution) will never be resolved. 

While it may never be resolved which one is better, the issue is entirely resolved for seL4. The decision has been made that the project is using `repo` and no new arguments have been brought forward that haven't been considered in that decision. If there are new aspects we'd be willing to discuss, but there is no point rehashing the same discussion every time.

I don't think it makes sense to use something else for the tutorials if people will have to deal with repo in any case to work with seL4. It is a hurdle, but it is not a huge hurdle.

**Comment by lsf37** [comment]
> @lsf37: how are the rules about the DCO, is `74560659+locriacyber@users.noreply.github.com` acceptable?

No, that is not acceptable. It needs to be something that identifies the person, it is a legal statement.

**Comment by locriacyber** [comment]
I hereby declare that I release all my past, present and future work related to the seL4 microkernel project (excluding work limited to a specific application using seL4) to the public domain, effective immediately.

A legal statement can be verbal. Plus, I don't have a public-facing email address. Plus, what if I'm in Russia?

**Comment by lsf37** [comment]
> I hereby declare that I release all my past, present and future work related to the seL4 microkernel project (excluding work limited to a specific application using seL4) to the public domain, effective immediately.

This is neither required not relevant. Please read the contribution guidelines and what the DCO actually declares.
 
> A legal statement can be verbal. Plus, I don't have a public-facing email address. Plus, what if I'm in Russia?

The requirement for contribution is a DCO. This requires an email address.

If you cannot provide one, that is fine, then we don't accept the contribution. We're certainly not going to jeopardise the legal copyright standing of a high-assurance microkernel because a contributor cannot or does not want to follow the rules for contributing to the repository.

**Comment by lsf37** [comment]
Closing this for inactivity. If someone else would like to take it up and is willing to follow the contribution requirements, please either re-open this one or open a new pull request.

**Pull request closed without merging**